signers/v2/v2.go

```diff
@@ -97,20 +97,35 @@ func (v *V2Signer) HashBytes(b []byte) string {
 func (v *V2Signer) CreateSignable(req *http.Request, authHeaders map[string]string, bodyhash string) []byte {
 	var b bytes.Buffer
 
+	// The uppercase HTTP request method e.g. "GET", "POST".
 	method := strings.ToUpper(req.Method)
-
 	b.WriteString(method)
 	b.WriteString("\n")
 
+	// The (lowercase) hostname, matching the HTTP "Host" request header field
+	// (including any port number).
 	b.WriteString(req.Host)
+
+	// If behind an ELB/Proxy, the port may not be included in the host field
+	if req.Header.Get("X-Forwarded-Port") != "" {
```

Comment:
What if the port is included? Probably need some regex query to find out if a port is added or not and if not, check if the X-Forwared-Port is available and if so, add it to the Host.